# Copyright 2026 The OpenXLA Authors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# ==============================================================================
Fix for b/405045790.

--- a/lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp
+++ b/lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp
@@ -361,9 +361,12 @@ static bool bwdFilter(Operation *op) {
       return false;
   }
 
+  // b/405045790: We don't want to propagate through the BroadcastOp because we
+  // probably don't care about the load before a broadcast as it would likely be
+  // small. This is just a heuristic to avoid a regression.
   return (op->hasTrait<OpTrait::Elementwise>() && isMemoryEffectFree(op)) ||
          isView(op) ||
-         isa<Fp4ToFpOp, LoadOp, DescriptorLoadLikeOpInterface, BroadcastOp,
+         isa<Fp4ToFpOp, LoadOp, DescriptorLoadLikeOpInterface,/*BroadcastOp,*/
              ConvertLayoutOp>(op);
 }
 

